src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -122,27 +121,35 @@ auto get_sshfs_exec_and_options(mp::SSHSession& session)
 // Split a path into existing and to-be-created parts.
 std::pair<std::string, std::string> get_path_split(mp::SSHSession& session, const std::string& target)
 {
-    QDir complete_path(QString::fromStdString(target));
-    QString absolute;
+    std::string complete_path;
 
-    if (complete_path.isRelative())
+    if ('~' == target[0])
     {
-        QString home = QString::fromStdString(run_cmd(session, "pwd")).trimmed();
-        absolute = home + '/' + complete_path.path();
+        complete_path = "~" + mp::utils::escape_all(target.substr(1, target.size() - 1));
     }
     else
     {
-        absolute = complete_path.path();
+        if ('/' != target[0])
+        {
+            complete_path = "~/" + mp::utils::escape_all(target);
+        }
+        else
+        {
+            complete_path = mp::utils::escape_all(target);
+        }
     }
 
-    QString existing =
-        QString::fromStdString(
-            run_cmd(session,
-                    fmt::format("sudo /bin/bash -c 'P=\"{}\"; while [ ! -d \"$P/\" ]; do P=${{P%/*}}; done; echo $P/'",
-                                absolute)))
-            .trimmed();
+    std::string existing =
+        run_cmd(session, fmt::format("sudo /bin/bash -c 'P={}; while [ ! -d $P/ ]; do P=${{P%/*}}; done; echo $P/'",
+                                     complete_path));
+    if ('\n' == existing.back())
+        existing.pop_back();
+
+    std::string missing = run_cmd(session, fmt::format("echo {} | cut -c {}-", complete_path, existing.size() + 1));
+    if ('\n' == missing.back())
+        missing.pop_back();
```

Comment:
Yes, that's indeed right. But the check is there because of the tests. Thinking a bit more, it's worth to pay a bit in the test code to make the code clearer here. (This check happens in a couple of places.)

Reply:
That, or maybe have a `trim_end` utility function that asserts it is actually a newline before popping off the back…

I would not like a bare `.pop_back()` here any more than this verbose approach… GAH.

Reply:
What disturbs me from creating an utility function or checking as I did is that the command will not end in newline in two cases: (i) it is a test, or (ii) there is an error somewhere. I can add a function for it because it is safer, but I vote also for adding the explicit `\n` at the end of each mock anster.

Reply:
Sure, which is why I suggested to actually `assert` that the trailing character _is_ a newline, it would mean the test would assert in Debug build mode (in CI), and fail if assertions are compiled out. And yes, all mocked answers should have a newline one way (explicit) or another (in `make_channel_read_return`).